src/format/parsed.rs

```diff
@@ -112,6 +112,7 @@ pub struct Parsed {
 
 /// Checks if `old` is either empty or has the same value to `new` (i.e. "consistent"),
 /// and if it is empty, set `old` to `new` as well.
+#[inline(always)]
```

Comment:
hm could we check what the bench results look like with just `inline` instead of `inline(always)`? My understanding is that that gives llvm more flexibility to decide whether to inline, although AIUI it's still just a hint anyway.

Reply:
I remeasured again and got these results:

No attribute:
```
test datetime::tests::bench_datetime_from_str            ... bench:         347 ns/iter (+/- 14)
test datetime::tests::bench_datetime_parse_from_rfc2822  ... bench:         197 ns/iter (+/- 121)
test datetime::tests::bench_datetime_parse_from_rfc3339  ... bench:         166 ns/iter (+/- 8)
```

With `#[inline]`:
```
test datetime::tests::bench_datetime_from_str            ... bench:         344 ns/iter (+/- 9)
test datetime::tests::bench_datetime_parse_from_rfc2822  ... bench:         192 ns/iter (+/- 11)
test datetime::tests::bench_datetime_parse_from_rfc3339  ... bench:         167 ns/iter (+/- 15)
```

With `#[inline(always)]`:
```
test datetime::tests::bench_datetime_from_str            ... bench:         369 ns/iter (+/- 33)
test datetime::tests::bench_datetime_parse_from_rfc2822  ... bench:         175 ns/iter (+/- 9)
test datetime::tests::bench_datetime_parse_from_rfc3339  ... bench:         138 ns/iter (+/- 5)
```

Interestingly the `bench_datetime_from_str` got worse, unlike in my previous measurement. The gains may be too random. I am fine with omitting this commit, it may be too risky to force inlining.